### ISSUE TITLE
PDFObjRef is not iterable

### ISSUE THREAD

**Issue by pietermarsman** [pdfminer/pdfminer.six#495]
**Bug report**

Copy of #471 (by @imochoa)

Sadly, I cannot upload the problematic PDFs due a non-disclosure agreement. I can however point out the issue and share my fix.

When trying to instantiate a PDFCIDFont object at:
https://github.com/pdfminer/pdfminer.six/blob/0b44f7771462363528c109f263276eb254c4fcd0/pdfminer/pdfinterp.py#L193

Where
```
self
<pdfminer.pdfinterp.PDFResourceManager at 0x7f0c80493048>

spec
{'Type': /'Font',
 'Subtype': /'CIDFontType2',
 'BaseFont': /'ISOCPEUR',
 'FontDescriptor': <PDFObjRef:5>,
 'CIDSystemInfo': <PDFObjRef:41>,
 'W': <PDFObjRef:13>,
 'Encoding': /'Identity-H'}
```

The execution ends up at PDFStream.decode with:
```
self
<PDFStream(7): raw=8963, {'Length1': 28048, 'Length': 8961, 'Filter': /'FlateDecode', 'DecodeParms': <PDFObjRef:39>}>
```

The origin of the bug is that in this case, `get_filters(self)`
https://github.com/pdfminer/pdfminer.six/blob/0b44f7771462363528c109f263276eb254c4fcd0/pdfminer/pdftypes.py#L258

returns:
```
 In[8]: filters
Out[8]: [(/'FlateDecode', <PDFObjRef:39>)]
```

As you can see, the second element of the first and only Tuple is a `PDFObjRef`, which is then saved to `params`  and fails a little later down the line when trying to evaluate `'Predictor' in params:`
https://github.com/pdfminer/pdfminer.six/blob/0b44f7771462363528c109f263276eb254c4fcd0/pdfminer/pdftypes.py#L297

I noticed that the default value of `params` is an empty dictionary, which effectively skips that check. So I extended the check to only continue if `params` was a dictionary:
https://github.com/imochoa/pdfminer.six/blob/2d996c9ae26c8a336711178a8afe3091e1140970/pdfminer/pdftypes.py#L297

I think the underlying error is the fact that `get_filters(self)` is returning a PDFObjRef instead of a dictionary. I've tried to find the exact origin, but I'm not too familiar with the project and couldn't pinpoint the exact issue. The furthest I got was that `PDFParser` was that the problematic `<PDFObjRef:39>`  value was being set at the dictionary in:
https://github.com/pdfminer/pdfminer.six/blob/0b44f7771462363528c109f263276eb254c4fcd0/pdfminer/pdfparser.py#L122

as:
```
dic['DecodeParms']
<PDFObjRef:39>
```

Since I couldn't prevent it from coming up, refining the check seemed like the next best option and it works well on the 2 problematic PDFs I have.

**Comment by imochoa** [comment]
Hello again :)

I found a PDF I can share where the issue is happening, in the text box at the bottom right where it says "PET BLACK"

[pdfminer_testpart.pdf](https://github.com/pdfminer/pdfminer.six/files/11768084/pdfminer_testpart.pdf)

**Comment by EvaSDK** [comment]
Jumping in as I had the same issue on some PII document that I would not be able to share.
The minimal code sample to reproduce is the following:

```python
from pdfminer.high_level import extract_text
extract_text("./pdfminer_testpart.pdf")
```

It should return:
```python
'8\n\n7\n\n6\n\n5\n\n4\n\n3\n\n2\n\n1\n\n150,00\n\n30,00\n\n(cid:72) 0,05 A\n\n0\n0\n,\n0\n2\n\n0\n0\n,\n8\n\n(cid:69) 0,05\n\n0\n0\n,\n0\n5\n\nA\n\nF\n\nE\n\nD\n\n20,00\n\n16,00\n\n+\n0,05\n15,00 - 0,00\n\nC\n\n0\n0\n,\n0\n4\n\n0\n0\n,\n0\n2\n\nR 1 8 , 0 0\n\nM12x1.75 - 6H\n\n0\n0\n,\n5\n4\n\nB\n\nA\n\n0\n0\n,\n6\n1\n(cid:142)\n\n0\n0\n,\n6\n1\n\n+\n0,50\n15,00 - 0,00\n\n60,00 (cid:66)0,02\n\n100,00 (cid:66)0,05\n\n132,00\n\n9\nH\n0\n1\n(cid:142)\n\n9\nH\n0\n1\n(cid:142)\n\n(cid:68) 0,1 A\n\n+\n0,00\n70,00 - 0,02\n\n50,00\n\n(cid:76) 0,1\n\n(cid:76) 0,1\n\n0\n0\n,\n5\n3\n\nF\n\nE\n\nD\n\nC\n\nB\n\nAllgemeintoleranzen\n\nMATERIAL\n\nDIN ISO 2768 - mK\n\nPET BLACK\n\nFINISH\n\nEloxieren (natur)\n\nRa 1,6\n\nDate\n29.03.2021\n\nName\nLucas Giering\n\nDrawn\n\nChecked\n\nStandard\n\nArretierungshilfe\n\nA\n\n1 \n\nA2\n\n8\n\n7\n\n6\n\n5\n\n4\n\nState\n\nChanges\n\nDate\n\nName\n\n3\n\n2\n\n1\n\n\x0c'
```

**Comment by pietermarsman** [comment]
Closed by #906
